tests/yast2_gui/yast2_firewall.pm

```diff
@@ -8,19 +8,26 @@
 # notice and this notice are preserved.  This file is offered as-is,
 # without any warranty.
 
-# Summary: Extend YaST2 Firewall UI test
+# Summary: YaST2 Firewall UI test checks verious configurations and settings of firewall
 # Make sure yast2 firewall can opened properly. Configurations can be changed and written correctly.
-# Maintainer: Max Lin <mlin@suse.com>
+# Maintainer: Zaoliang Luo <zluo@suse.com>
 
 use base "y2x11test";
 use strict;
 use testapi;
+use utils;
 
 sub run() {
     my $self = shift;
 
-    $self->launch_yast2_module_x11('firewall');
+    #	make sure http server got installed on openSUSE TW, otherwise the needle yast2_firewall_service_http cannot be matched
+    if (check_var('DISTRI', 'opensuse')) {
+        select_console 'root-console';
+        zypper_call('in yast2-http-server apache2 apache2-prefork');
```

Comment:
wouldn't that be a good idea on both openSUSE as well as SLE? I assume the packages are called the same. On the other hand: It might be a product bug when they behave that different.